core/controller/src/main/scala/whisk/core/controller/Controller.scala

```diff
@@ -171,8 +173,8 @@ object Controller {
 
     def main(args: Array[String]): Unit = {
         implicit val actorSystem = ActorSystem("controller-actor-system")
-        implicit val logger = new AkkaLogging(akka.event.Logging.getLogger(actorSystem, this))
-
+        implicit val logger = new ZipkinLogging(new AkkaLogging(akka.event.Logging.getLogger(actorSystem, this)))
```

Comment:
On logs, @jeremiaswerner and his colleague Vadim have a prototype which I've suggested they share on the dev list. I expect they will soon. It follows this mode where the logs can be tee'd to Elastic. One point of note on this though is that a feature of openwhisk we should not give up is rapid access to action logs (within 1-2s). This is very helpful especially when doing rapid development and debugging. But in general, yes. 

On the database - here it is a bit more nuanced. We had started with a plugable model (there's an abstract database interface) but we are tied to couchdb and it will require quite a bit of work to support say a relational datastore model. It fits openwhisk well as we are storing relational data though.

On leveraging HTTP over Kafka - the point of Kafka is that it gives you the ability to persist message, recover, and even replay, at scale. If you replace it you will need to roll a RAS story for it. That's not to say it's not possible. We have discussed for example using akka messaging but then you have to roll persistence and recovery.

So I think where it makes sense to have plugin modes, yes.